File: src/main/java/com/ggemo/bilidanmakuclient/http/request/DanmakuServerConfRequest.java
package com.ggemo.bilidanmakuclient.http.request;

import com.ggemo.bilidanmakuclient.http.MyHttpClient;
import com.ggemo.bilidanmakuclient.http.response.impl.DanmakuServerConfResponse;
import com.ggemo.bilidanmakuclient.http.response.impl.RoomInitResponse;
import org.apache.http.HttpEntity;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.util.EntityUtils;

import java.io.IOException;

public class DanmakuServerConfRequest implements Request {
    private static final String DANMAKU_SERVER_CONF_URL_TEMPLATE = "https://api.live.bilibili.com/room/v1/Danmu/getConf?id=%d";

    private MyHttpClient httpClient;


    public DanmakuServerConfRequest() {
        this.httpClient = MyHttpClient.getInstance();
    }

    public DanmakuServerConfResponse request(long roomId) throws IOException {
        HttpGet httpGet = new HttpGet(String.format(DANMAKU_SERVER_CONF_URL_TEMPLATE, roomId));
        CloseableHttpResponse httpResponse = this.httpClient.request(httpGet);
        HttpEntity responseEntity = null;
        try (httpResponse) {
            responseEntity = httpResponse.getEntity();
            return DanmakuServerConfResponse.parse(EntityUtils.toString(responseEntity));
        } finally {
            EntityUtils.consume(responseEntity);
        }
    }
}

File: blive-danmaku-client-jvm/src/main/kotlin/com/ggemo/va/blivedanmakuclient/inner/httpclient/request/DanmakuServerConfRequest.kt
package com.ggemo.va.bilidanmakuclient.http.request

import com.ggemo.va.bilidanmakuclient.http.MyHttpClient
import kotlin.Throws
import java.io.IOException
import org.apache.http.client.methods.HttpGet
import org.apache.http.HttpEntity
import org.apache.http.util.EntityUtils
import com.ggemo.va.bilidanmakuclient.http.response.impl.DanmakuServerConfResponse

class DanmakuServerConfRequest : Request {
    private val httpClient: MyHttpClient
    @Throws(IOException::class)
    fun request(roomId: Long): DanmakuServerConfResponse {
        val httpGet = HttpGet(String.format(DANMAKU_SERVER_CONF_URL_TEMPLATE, roomId))
        val httpResponse = httpClient.request(httpGet)
        var responseEntity: HttpEntity? = null
        try {
            httpResponse.use {
                responseEntity = httpResponse.entity
                return DanmakuServerConfResponse.Companion.parse(EntityUtils.toString(responseEntity))
            }
        } finally {
            EntityUtils.consume(responseEntity)
        }
    }

    companion object {
        private const val DANMAKU_SERVER_CONF_URL_TEMPLATE =
            "https://api.live.bilibili.com/room/v1/Danmu/getConf?id=%d&platform=pc&player=web"
    }

    init {
        httpClient = MyHttpClient.instance
    }
}